frame/vault/src/lib.rs

```diff
@@ -714,6 +734,11 @@ pub mod pallet {
 				.fold(T::Balance::zero(), |sum, item| sum + item.balance);
 			Ok(owned + outstanding)
 		}
+
+		/// Tries to fetch a stored [VaultInfo] through its index.
+		fn vault_info(vault_idx: &VaultIndex) -> Result<VaultInfo<T>, DispatchError> {
+			Ok(Vaults::<T>::try_get(vault_idx).map_err(|_err| Error::<T>::VaultDoesNotExist)?)
+		}
```

Comment:
De-duplicates several occurrences of `Vaults::<T>::try_get(&vault_id).map_err(|_| Error::<T>::VaultDoesNotExist)?;` that were scattered across the crate